astropy/units/equivalencies.py

```diff
@@ -503,9 +503,31 @@ def from_vel_en(x):
         return resten * ((1-voverc) / (1+(voverc)))**0.5
 
     return Equivalency([(si.Hz, si.km/si.s, to_vel_freq, from_vel_freq),
-            (si.AA, si.km/si.s, to_vel_wav, from_vel_wav),
-            (si.eV, si.km/si.s, to_vel_en, from_vel_en),
-            ], "doppler_relativistic", {'rest': rest})
+                        (si.AA, si.km/si.s, to_vel_wav, from_vel_wav),
+                        (si.eV, si.km/si.s, to_vel_en, from_vel_en),
+                        ], "doppler_relativistic", {'rest': rest})
+
+
+def doppler_redshift():
+    """
+    Returns the equivalence between Doppler redshift (unitless
+    or `astropy.cosmology.units.redshift`) and radial velocity.
+    """
+    from astropy.cosmology.units import redshift
+    rv_unit = si.km / si.s
+    C_KMS = _si.c.to_value(rv_unit)
+
+    def convert_z_to_rv(z):
+        zponesq = (1 + z) ** 2
+        return (C_KMS * (zponesq - 1) / (zponesq + 1))
```

Comment:
```suggestion
    def convert_z_to_rv(z):
        if hasattr(z, "unit") and z.unit == redshift:
            raise UnitError("cannot interpret a cosmological redshift as a Doppler velocity.")
        zponesq = (1 + z) ** 2
        return (C_KMS * (zponesq - 1) / (zponesq + 1))
```

Reply:
This might be a little overkill. If I simply omit `redshift` altogether from the list, it raises `UnitConversionError` like any other incompatible unit. I already have a commit ready to go. Will push it out in a few mins.

Reply:
This change is incorrect - the conversion functions get values in the equivalency unit, not quantity, so no need to check for units, etc.

Reply:
In other words, @pllim is right, just removing `redshift` from the equivalency is good enough.